### PR TITLE
Formatting and typing cleanup

### PR DESCRIPTION
**Pull request by ecraig12345**
Some typing and formatting fixes I noticed would be helpful while doing other work on beachball:
- Enable `noImplicitAny` and `noUnusedLocals` and make related typing fixes
- Update prettier to handle new TS syntax
- Fix prettier commands and run them
- Manual formatting fixes: add blank lines; remove empty param tags
- Use `fs-extra` (and its promise-based async methods) everywhere

### CODE REVIEW

File: packages/beachball/src/bump/setDependentsInBumpInfo.ts [hunk 1]
```diff
@@ -19,8 +19,9 @@ export function setDependentsInBumpInfo(bumpInfo: BumpInfo): void {
     const info = packageInfos[pkgName];
     const depTypes = ['dependencies', 'devDependencies', 'peerDependencies'];
     depTypes.forEach(depType => {
-      if (info[depType]) {
-        for (let [dep, _] of Object.entries(info[depType])) {
+      const deps: PackageDeps | undefined = (info as any)[depType];
+      if (deps) {
+        for (let dep of Object.keys(deps)) {
```
Comment (xugao) [inline packages/beachball/src/bump/setDependentsInBumpInfo.ts:24]: directly use `for .. in`: `for (let depPackage in deps)`

Comment (ecraig12345) [reply packages/beachball/src/bump/setDependentsInBumpInfo.ts:24]: I thought that was still considered bad practice due to `for..in` including non-own properties

Comment (xugao) [reply packages/beachball/src/bump/setDependentsInBumpInfo.ts:24]: hm i never heard of it as bad practice. in this use case, i dont think there is a real concern?

Comment (ecraig12345) [reply packages/beachball/src/bump/setDependentsInBumpInfo.ts:24]: We have a lint rule against it in Fabric (unless accompanied by a hasOwnProperty check), and my understanding is that it's a common lint rule in other projects too. Object.keys is much more commonly used than for..in across both beachball and oufr.

**Pull request merged**